system-tests/lib/cre/don/config/config.go

```diff
@@ -4,140 +4,342 @@ import (
-	case 1:
-		bootstrapNode := bootstrapNodes[0]
+	existingConfig.P2P = coretoml.P2P{
+		V2: coretoml.P2PV2{
+			Enabled:              ptr.Ptr(true),
+			ListenAddresses:      ptr.Ptr([]string{"0.0.0.0:" + fmt.Sprint(ocrPeeringData.Port)}),
+			DefaultBootstrappers: ptr.Ptr([]commontypes.BootstrapperLocator{*ocrBoostrapperLocator}),
+		},
+	}
 
-		donBootstrapNodePeerID, err = node.ToP2PID(bootstrapNode, node.KeyExtractingTransformFn)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to get bootstrap node peer ID")
+	capabilitiesBootstrapperLocator, capabilitiesBErr := commontypes.NewBootstrapperLocator(capabilitiesPeeringData.GlobalBootstraperPeerID, []string{capabilitiesPeeringData.GlobalBootstraperHost + ":" + fmt.Sprint(capabilitiesPeeringData.Port)})
+	if capabilitiesBErr != nil {
+		return existingConfig, errors.Wrap(capabilitiesBErr, "failed to create capabilities peering bootstrapper locator")
+	}
+
+	existingConfig.Capabilities = coretoml.Capabilities{
+		Peering: coretoml.P2P{
+			V2: coretoml.P2PV2{
+				Enabled:              ptr.Ptr(true),
+				ListenAddresses:      ptr.Ptr([]string{"0.0.0.0:" + fmt.Sprint(capabilitiesPeeringData.Port)}),
+				DefaultBootstrappers: ptr.Ptr([]commontypes.BootstrapperLocator{*capabilitiesBootstrapperLocator}),
+			},
+		},
+	}
+
+	for _, evmChain := range commonInputs.evmChains {
+		existingConfig.EVM = append(existingConfig.EVM, &evmconfigtoml.EVMConfig{
+			ChainID: chainlinkbig.New(big.NewInt(libc.MustSafeInt64(evmChain.ChainID))),
+			Chain: evmconfigtoml.Chain{
+				AutoCreateKey: ptr.Ptr(false),
+			},
+			Nodes: []*evmconfigtoml.Node{
+				{
+					Name:    ptr.Ptr(evmChain.Name),
+					WSURL:   commonconfig.MustParseURL(evmChain.WSRPC),
+					HTTPURL: commonconfig.MustParseURL(evmChain.HTTPRPC),
+				},
+			},
+		})
+	}
+
+	if commonInputs.solanaChain != nil {
+		existingConfig.Solana = append(existingConfig.Solana, &solcfg.TOMLConfig{
+			ChainID: ptr.Ptr(commonInputs.solanaChain.ChainID),
+			Enabled: ptr.Ptr(true),
+			Nodes: solcfg.Nodes{
+				{
+					Name: ptr.Ptr(commonInputs.solanaChain.Name),
+					URL:  commonconfig.MustParseURL(commonInputs.solanaChain.NodeURL),
+				},
+			},
+		})
+	}
+
+	existingConfig.Capabilities.ExternalRegistry = coretoml.ExternalRegistry{
+		Address:   ptr.Ptr(commonInputs.capabilitiesRegistryAddress.Hex()),
+		NetworkID: ptr.Ptr("evm"),
+		ChainID:   ptr.Ptr(fmt.Sprint(commonInputs.registryChainID)),
+	}
+
+	if flags.HasFlag(donFlags, cre.WorkflowDON) {
+		existingConfig.Capabilities.WorkflowRegistry = coretoml.WorkflowRegistry{
+			Address:   ptr.Ptr(commonInputs.workflowRegistryAddress.Hex()),
+			NetworkID: ptr.Ptr("evm"),
+			ChainID:   ptr.Ptr(fmt.Sprint(commonInputs.registryChainID)),
 		}
+	}
 
-		for _, label := range bootstrapNode.Labels {
-			if label.Key == node.HostLabelKey {
-				donBootstrapNodeHost = label.Value
+	if flags.HasFlag(donFlags, cre.WorkflowDON) || don.NodeNeedsAnyGateway(donFlags) {
```

Comment:
>don.NodeNeedsAnyGateway(donFlags)

Also a bit smelly -- looks like this should be moved to capability configuration since the implementation looks like this:
```
	return flags.HasFlag(nodeFlags, cre.CustomComputeCapability) ||
		flags.HasFlag(nodeFlags, cre.WebAPITriggerCapability) ||
		flags.HasFlag(nodeFlags, cre.WebAPITargetCapability) ||
		flags.HasFlag(nodeFlags, cre.VaultCapability) ||
		flags.HasFlag(nodeFlags, cre.HTTPActionCapability) ||
		flags.HasFlag(nodeFlags, cre.HTTPTriggerCapability)
```

(Can be deferred as a separate refactor)

Reply:
How do you envision that?

Every capability having a function `needsGateway() bool` and then we'd iterate over all capabilities that the DON has to call it?

Reply:
next PR, ok? added to the list